common/src/stack/report-system/command/report/system/tests/test_backend_link.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+import paramiko
+import socket
+from stack import api
+
+@pytest.mark.parametrize('backend',
+[host['host'] for host in  api.Call('list host', args=['a:backend'])]
+)
+class TestLinkUp:
+
+	# See if the interfaces stacki configures are actually up on the hosts
+
+		def test_link_status(self, backend):
+			errors = []
+			backend_ssh = paramiko.SSHClient()
+			backend_ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+			# Get all interfaces for this host
+			interfaces = [
+				host['interface'] for host in api.Call('list host interface ', args=['a:backend']) if host['host'] == backend
+			]
+
+			try:
+				backend_ssh.connect(hostname=f'{backend}')
+
+			except (paramiko.SSHException, socket.error):
+				pytest.skip(f'Could not ssh into host {backend}')
+
+			for link in interfaces:
+
+				# Run ethtool on the interface
+				stdin, stdout, stderr = backend_ssh.exec_command(f'ethtool {link}')
+				interface_data = stdout.readlines()
+
+				# Flag which is set to true if ethtool reports the link is up
+				link_up = False
+
+				for line in interface_data:
+					if 'Link detected: yes' in line.rstrip():
```

Comment:
I don't think the `rstrip` is needed if this is just checking whether the string is in the other string. It might be good to use `lower` on the line, however, and perform a case insensitive compare.

Reply:
I don't need rstrip to check if there is a sub string true, I was trying to get rid of the new line chars the output creates.

Reply:
I don't see `line` being used to output information to the user? Maybe I'm missing something?

Reply:
In Python strings are immutable. So, line.rstrip() returns a new string. If you want to use that string elsewhere, you'll need to assign it to something.

Reply:
Ya I realize it's not really necessary as I'm just parsing the string vs printing it out again.